lib/ansible/modules/packaging/language/pip.py

```diff
@@ -235,6 +241,72 @@
                              'pip': 'import pkg_resources; print(pkg_resources.get_distribution("pip").version)'}
 
 
+class Distribution(object):
+    def __init__(self, name_string, version_string=None):
+        if not name_string.startswith(('svn+', 'git+', 'hg+', 'bzr+', 'file:')):
+            self._plain_distribution = True
+            if version_string:
+                version_string = version_string.lstrip()
+                separator = '==' if version_string[0].isdigit() else ' '
+                name_string = separator.join((name_string, version_string))
+            self._requirement = Requirement.parse(name_string)
+            self._distribution_name = self._requirement.name.lower()
+        else:
+            self._plain_distribution = False
+            self._distribution_name = name_string
+
+    @property
+    def has_version_specifier(self):
+        if self._plain_distribution:
+            return bool(self._requirement.specs)
+        return False
+
+    @property
+    def distribution_name(self):
+        return self._distribution_name
+
+    @distribution_name.setter
```

Comment:
Since you don't use setter you probably don't need it.

Reply:
in __init__ I need a `self.__distribution_name = self._requirement.name.lower()`

Reply:
ah, that error happened when I didn't rename the `distribution_name` to `_distribution_name`. You are right